### PR TITLE
WT-9614 Improve required compiler and python support in cmake/doc

### PR DESCRIPTION
**Pull request by mickgraham-mongodb**
Updating cmake to default python enable/disable based on presence. Requiring C11 standard for compiler. Improving doc around tool requirements and cmake features.

### CODE REVIEW

File: src/docs/build-posix.dox [hunk 1]
```diff
@@ -102,7 +103,7 @@ Configure WiredTiger to sleep and wait for a debugger to attach on failure.
 <b>DO NOT</b> configure this option in production environments.
 
 @par \c -DHAVE_DIAGNOSTIC=1
-Configure WiredTiger to perform various run-time diagnostic tests.
+Configure WiredTiger to perform various run-time diagnostic tests (enabled by default).
```
Comment (lukech) [inline src/docs/build-posix.dox:106]: Though not related to this PR, it does not feels super safe having the default on for diagnostics. Users need to consciously remember setting `-DHAVE_DIAGNOSTIC=0` for their production builds.

File: test/evergreen.yml [hunk 2]
```diff
@@ -4363,6 +4363,8 @@ buildvariants:
   - windows-64-vs2017-test
   expansions:
     python_binary: '/cygdrive/c/Python39/python'
+    configure_env_vars:
```
Comment (lukech) [inline test/evergreen.yml:4366]: Thanks for fixing this!

**Review by lukech — APPROVED** [review]
LGTM

File: src/docs/build-posix.dox [hunk 3]
```diff
@@ -10,6 +10,7 @@ To build from the WiredTiger GitHub repository requires
 <a href="https://gcc.gnu.org/">GCC 8.5 or later</a> or
 <a href="https://clang.llvm.org/">Clang 7.01 or later</a>, and
 <a href="https://cmake.org/">CMake 3.10.0 or later</a>, and
+<a href="https://python.org/">Python 3 or later</a> (if building to support the Python API or running the tests), and
```
[the remark this answers is not in the document]
Comment (mickgraham-mongodb) [reply src/docs/build-posix.dox]: Fixed

File: src/docs/build-posix.dox (hunk 1 above)
Comment (mickgraham-mongodb) [reply src/docs/build-posix.dox]: Cleaned it up, we enable it by default for non Release build types.

**Pull request merged**